packages/esm-patient-vitals-app/src/vitals/vitals.resource.tsx

```diff
@@ -1,81 +1,78 @@
-    respiratoryRate: concepts.respiratoryRateUuid,
+    systolicBloodPressure: config.concepts.systolicBloodPressureUuid,
+    diastolicBloodPressure: config.concepts.diastolicBloodPressureUuid,
+    pulse: config.concepts.pulseUuid,
+    temperature: config.concepts.temperatureUuid,
+    oxygenSaturation: config.concepts.oxygenSaturationUuid,
+    height: config.concepts.heightUuid,
+    weight: config.concepts.weightUuid,
+    respiratoryRate: config.concepts.respiratoryRateUuid,
   };
 
-  return openmrsObservableFetch<VitalsFetchResponse>(
-    `${fhirBaseUrl}/Observation?subject:Patient=${patientID}&code=` +
+  const { data, error, isValidating } = useSWR<{ data: VitalsFetchResponse }, Error>(
+    `${fhirBaseUrl}/Observation?subject:Patient=${patientUuid}&code=` +
       Object.values(vitalsConcepts).join(',') +
       '&_summary=data&_sort=-date' +
-      `&_count=${pageSize}`,
-  ).pipe(
-    map(({ data }) => data.entry),
-    map((entries) => entries?.map((entry) => entry.resource) ?? []),
-    map((vitals) => {
-      return formatVitals(
-        filterByConceptUuid(vitals, concepts.systolicBloodPressureUuid),
-        filterByConceptUuid(vitals, concepts.diastolicBloodPressureUuid),
-        filterByConceptUuid(vitals, concepts.pulseUuid),
-        filterByConceptUuid(vitals, concepts.temperatureUuid),
-        filterByConceptUuid(vitals, concepts.oxygenSaturationUuid),
-        filterByConceptUuid(vitals, concepts.heightUuid),
-        filterByConceptUuid(vitals, concepts.weightUuid),
-        filterByConceptUuid(vitals, concepts.respiratoryRateUuid),
-      );
-    }),
+      `&_count=${pageSize}
+  `,
+    openmrsFetch,
+  );
+
+  const observations = data?.data?.total > 0 ? data.data?.entry?.map((entry) => entry.resource ?? []) : null;
+
+  const systolicBloodPressureData = observations?.filter((obs) =>
+    obs.code.coding.some((sys) => sys.code === config.concepts.systolicBloodPressureUuid),
+  );
+  const diastolicBloodPressureData = observations?.filter((obs) =>
+    obs.code.coding.some((sys) => sys.code === config.concepts.diastolicBloodPressureUuid),
+  );
+  const pulseData = observations?.filter((obs) =>
+    obs.code.coding.some((sys) => sys.code === config.concepts.pulseUuid),
+  );
+  const temperatureData = observations?.filter((obs) =>
+    obs.code.coding.some((sys) => sys.code === config.concepts.temperatureUuid),
   );
+  const oxygenSaturationData = observations?.filter((obs) =>
+    obs.code.coding.some((sys) => sys.code === config.concepts.oxygenSaturationUuid),
+  );
+  const heightData = observations?.filter((obs) =>
+    obs.code.coding.some((sys) => sys.code === config.concepts.heightUuid),
+  );
+  const weightData = observations?.filter((obs) =>
+    obs.code.coding.some((sys) => sys.code === config.concepts.weightUuid),
+  );
+  const respiratoryRateData = observations?.filter((obs) =>
+    obs.code.coding.some((sys) => sys.code === config.concepts.respiratoryRateUuid),
+  );
```

Comment:
I think this was better as it was before. That `filterByConceptUuid` function could even be defined as an arrow function right on top of these lines, so it's all right there.